src/Npgsql/NpgsqlTypes/NpgsqlDate.cs

```diff
@@ -10,6 +10,9 @@
 namespace NpgsqlTypes
 {
     [Serializable]
+    [Obsolete("This type will be removed from the next release. Please use the System.DateTime"
```

Comment:
Suggestion: we may want to open a new issue, called "New PostgreSQL-specific date/time types?", rather than pointing to #2009 - first, that issue will be closed by this PR, and it would also be good to have a blank, fresh issue with a clear description of why those types may be needed.